developer_guides/debugability/perf-counters/index.rst

```diff
@@ -0,0 +1,62 @@
+.. _dbg-perf-counters:
+
+Performance Counters
+####################
+
+Firmware can be configured to trace performance counters for each processing
+component. Each performance trace entry include: 
+
+- component UUID
+
+- peak platform & cpu timer ticks consumed during component's copy processing
+
+.. note::
+  Performance timestamp macros use both the platform timer and cpu timer in case
+  the latter is not always-running-timer.
+
+Performance Counters usage
+**************************
+
+Currently performance counters can be only enabled statically during FW build in
+one of two ways:
+
+#. Select "Performance counter" from "Debug" menu using make menuconfig
+
+#. Add "CONFIG_PERFORMANCE_COUNTERS=y" to specific FW config (e.g. tigerlake_defconfig)
+
+
+Once the the performance counters are enabled they are logged periodically for
+each active component with the pipeline period frequency.
+
+Example
+*******
+
+Performance counter trace example:
+
+   .. code-block:: bash
+
+      [ 8481257.031250] ( 51.562500) c0 demux 1.2 src/audio/pipeline.c:206 perf comp_copy peak plat 782 cpu 8136
+
+``demux 1.2`` - processing component
+
+``plat 782`` - peak platform cycles consumed
+
+``cpu 8136`` - peak CPU cycles consumed
+
+MCPS calculation
+----------------
+
+The equation below illustrates how to calculate component MCPS (million cycles
+per second) consumption.
+
+   .. code-block:: bash
+
+      MCPS = cpu_ticks / (pipeline_period[s] * 10^6)
+      
+      // for common pipeline_period = 1ms it can be simplified to
+      MCPS = cpu_ticks / 1000
+
+In the trace example above the cpu_ticks = 8136, the pipeline_period is 1ms so
+the demux consumption equals 8,136 MCPS
```

Comment:
In the trace example above, *cpu_ticks* is 8136, *pipeline_period* is 1 ms, so
the demux consumption equals 8,136 MCPS.